.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,4 +1,39 @@
-### Description
+## Description
+Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any.
 
-### Test Plan
-<!-- Please provide us with clear details for verifying that your changes work. -->
+## Type of Change
+- [ ] New feature
+- [ ] Bug fix
+- [ ] Performance improvement
+- [ ] Refactoring
+- [ ] Dependency update
+- [ ] Documentation update
+
+## Which Component or System Does This Change Impact?
+- [ ] Validator Node
+- [ ] Full Node (API, Indexer, etc.)
+- [ ] Move/Aptos Virtual Machine
+- [ ] Aptos Framework
+- [ ] Aptos CLI/SDK
+- [ ] Developer Infrastructure
+- [ ] Other (specify)
+
+## How Has This Been Tested?
+Please ensure that the functionality introduced by this change is well tested and verified to work as expected. Ensure tests cover both happy and unhappy paths.
+
+- [ ] Test A
+- [ ] Test B
+
+## Key Areas to Review
+- Identify any critical parts of the code that require special attention or understanding. Explain why these parts are crucial to the functionality or architecture of the project.
+- Point out any areas where complex logic has been implemented. Provide a brief explanation of the logic and your approach to make it easier for reviewers to follow.
+- Highlight any areas where you are particularly concerned or unsure about the code's impact on the project. This can include potential performance or security issues, or compatibility with existing features.
+
+## Checklist
+- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I tested both happy and unhappy path of the functionality
+- [ ] I have made corresponding changes to the documentation
+
+Thank you for your contribution!
```

Comment:
This should be a comment at best, as this will be perspective of the author itself.

Reply:
technically a lot of the content should be a comment, but I didn't like adding `<!-- -->` everywhere. no strong feelings though

Reply:
added comments